base/timing.jl

```diff
@@ -115,35 +124,39 @@ function format_bytes(bytes) # also used by InteractiveUtils
     end
 end
 
-function time_print(elapsedtime, bytes=0, gctime=0, allocs=0, compile_time=0)
-    timestr = Ryu.writefixed(Float64(elapsedtime/1e9), 6)
-    length(timestr) < 10 && print(" "^(10 - length(timestr)))
-    print(timestr, " seconds")
-    parens = bytes != 0 || allocs != 0 || gctime > 0 || compile_time > 0
-    parens && print(" (")
-    if bytes != 0 || allocs != 0
-        allocs, ma = prettyprint_getunits(allocs, length(_cnt_units), Int64(1000))
-        if ma == 1
-            print(Int(allocs), _cnt_units[ma], allocs==1 ? " allocation: " : " allocations: ")
-        else
-            print(Ryu.writefixed(Float64(allocs), 2), _cnt_units[ma], " allocations: ")
-        end
-        print(format_bytes(bytes))
-    end
-    if gctime > 0
+const print_lock = ReentrantLock()
+function time_print(elapsedtime, bytes=0, gctime=0, allocs=0, compile_time=0, newline=false)
+    lock(print_lock) do
```

Comment:
This could use `lock(stdout) do` instead.

Reply:
Given https://github.com/JuliaLang/julia/blob/5cd07f84d986f3a2534401a560e38d29e5e5703c/test/runtests.jl#L113 I assumed that there are times with `stdout` can't be locked. And I thought that if I implemented it the same as the runtests `print_lock` it would need to be evaluated at runtime, so I thought I'd just keep it simpler with a `ReentrantLock`?

Reply:
Creating a new lock here can make it much harder to use this function while avoiding creating deadlock. No function that this calls may then safely acquire the `lock(stdout)` in that case.

Reply:
ok, I'll change strategy and just collate the text into a string and print it once within the function. That should help tread-clashing report tidyness a fair bit

Reply:
Updated to collate a string and print once now, with no print lock